app/controllers/controller.py

```diff
@@ -20,30 +21,47 @@ def clear_image_cache(image_path):
     def pad(s):
         return s + (16 - len(s) % 16) * "{"
 
-    # for prefix in ['http://www.bethel.edu', 'https://www.bethel.edu',
-    #                'http://staging.bethel.edu', 'https://staging.bethel.edu',
-    #                'http://thumbor.bethel.edu', 'https://thumbor.bethel.edu']:
-    path = image_path
-    # digest = hashlib.sha1(path.encode('utf-8')).hexdigest()
-    # url = "%s/%s" % (generated_url, hashlib.md5(image).hexdigest())
+    def path_on_filesystem(path):
+        path = re.sub("\:", "%3A", path)
+        digest = hashlib.sha1(path.encode('utf-8')).hexdigest()
+        return "%s/%s/%s" % (
+            app.config['THUMBOR_STORAGE_LOCATION'].rstrip('/'),
+            digest[:2],
+            digest[2:]
+        )
 
-    path = "%s/%s/%s" % (app.config['THUMBOR_STORAGE_LOCATION'].rstrip('/'), digest[:2], digest[2:])
-    resp.append(digest)
 
-    # remove the file at the path
-    # call(['rm', path])
+    for prefix in ['http://www.bethel.edu', 'https://www.bethel.edu',
+                   'http://staging.bethel.edu', 'https://staging.bethel.edu']:
+        path = prefix + image_path
+        resp.append(path)
+        encrypted_path = path_on_filesystem(path)
+        resp.append(encrypted_path)
 
-    print resp
+        # remove the file at the path
+        subprocess.call(['rm', encrypted_path])
 
-    # now the result storage
-    file_name = image_path.split('/')[-1]
-    matches = []
-    for root, dirnames, filenames in os.walk(app.config['THUMBOR_RESULT_STORAGE_LOCATION']):
-        for filename in fnmatch.filter(filenames, file_name):
-            matches.append(os.path.join(root, filename))
-    for match in matches:
-        call(['rm', match])
+    # For local testing
+    # cmd = 'find /Users/josiahtillman/Desktop -wholename "*/TestFolder/*"'
+    cmd = 'find ' + app.config['THUMBOR_RESULT_STORAGE_LOCATION'] + ' -wholename "*/smart/*' + image_path + '"'
```

Comment:
Sorry to nitpick...

This variable assignment might be more legible if this was changed to use string formatting:
```
cmd = 'find %s -wholename "*/smart/*%s"' % (app.config['THUMBOR_RESULT_STORAGE_LOCATION'], image_path)
``

Reply:
There are a few other places where formatting could be used in that file, also. This one should really get changed, but the other ones aren't as necessary